### PR TITLE
RNN docs

### PR DESCRIPTION
**Pull request by jeremiedb**
This PR is to augment the documentation around the recurrence structure to address frequent questions raised about how to handle the recurrent tools for various use cases.

This is presented as an additional component to the current Recurrent doc. I'm unclear if it would be preferable to attempt to blend the current docs into a single page on recurrence?

### CODE REVIEW

**Review by ToucheSir — COMMENTED** [review]
This is great! I vote for merging with the current RNN page, as having a "continued" page would be somewhat awkward for navigation.

**Comment by jeremiedb** [comment]
@ToucheSir Makes much sense! So I'll rework to merge some the more detailed Recurrent cell logic from the original docs into the new step expanded explanations.

**Comment by jeremiedb** [comment]
@CarloLucibello / @DhairyaLGandhi  Update to the recurrent docs would be now ready for review.

**Comment by jeremiedb** [comment]
@ToucheSir Just bumping to avoid it falls in a crack. Is there anything else needed for merge?

File: docs/src/models/recurrence.md [hunk 1]
```diff
@@ -1,34 +1,19 @@
 # Recurrent Models
 
-## Recurrent Cells
+## Recurrent cells
 
-In the simple feedforward case, our model `m` is a simple function from various inputs `xᵢ` to predictions `yᵢ`. (For example, each `x` might be an MNIST digit and each `y` a digit label.) Each prediction is completely independent of any others, and using the same `x` will always produce the same `y`.
+To introduce Flux's recurrence functionnalities, we will consider the following vanilla recurrent neural network structure:
```
Comment (CarloLucibello) [inline docs/src/models/recurrence.md:5]:
```suggestion
To introduce Flux's recurrence functionalities, we will consider the following vanilla recurrent neural network structure:
```

**Review by CarloLucibello — APPROVED** [review]

**Comment by CarloLucibello** [comment]
best documentation page we'll have in Flux, thanks!

bors r+

**Pull request merged**